File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        output=""
        max_len=0
        for i in range(len(s)):
            if s[i] in output:
                idx=output.index(s[i])
                output=output[idx+1:]
            output+=s[i]
            max_len=max(max_len,len(output))
        return max_len

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> set=new HashSet<>();
        int maxlen=0;
        int left=0;
        for(int right=0;right<s.length();right++){
            char curr=s.charAt(right);
            while(set.contains(curr)){
                set.remove(s.charAt(left));
                left++;
            }
            set.add(curr);
            maxlen=Math.max(maxlen,right-left+1);
        }
        return maxlen;
    }
}